### DISCUSSION TITLE
enh: "413 Request Entity Too Large" not handled well, response HTML treated as JSON

### DISCUSSION THREAD

**Discussion by darkvertex** [open-webui/open-webui#6429] [General]
# Bug Report

## Description

**Bug Summary:**
The OpenAI /completions proxy endpoint is responding with HTTP status code `413 Request Entity Too Large`:

```html
<html>
<head><title>413 Request Entity Too Large</title></head>
<body>
<center><h1>413 Request Entity Too Large</h1></center>
<hr><center>nginx/1.26.1</center>
</body>
</html>
<!-- a padding to disable MSIE and Chrome friendly error page -->
<!-- a padding to disable MSIE and Chrome friendly error page -->
<!-- a padding to disable MSIE and Chrome friendly error page -->
<!-- a padding to disable MSIE and Chrome friendly error page -->
<!-- a padding to disable MSIE and Chrome friendly error page -->
<!-- a padding to disable MSIE and Chrome friendly error page -->
```

**Steps to Reproduce:**
Have a conversation in Call mode using the webcam feed and do a few pictures. After ~5-10 it should complain.

**Expected Behavior:**
I'd expect the UI to render a more correct error and maybe offer to continue without the full conversation context.

**Actual Behavior:**

It tries to interpret the raw HTML:
<img width="383" alt="image" src="https://github.com/open-webui/open-webui/assets/941331/df12d096-d8ef-430f-990b-499b646e88e7">

## Environment

- **Open WebUI Version:** v0.3.5
- **Ollama (if applicable):** 0.1.44

- **Operating System:** Windows 11

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [X] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
```
  
        Failed to load resource: the server responded with a status of 413 ()
Chat.svelte:1099  SyntaxError: Unexpected token '<', "<html>
<h"... is not valid JSON
P @ Chat.svelte:1099
CallOverlay.svelte:506 Audio monitoring and playing stopped for message ID ea7f53d0-b8df-40c1-8102-202536b2ecc0
02fc276d-ae2a-45cd-9249-410da7b04522:1  [Deprecation] -ms-high-contrast-adjust is in the process of being deprecated. Please see https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/ for tips on updating to the new Forced Colors Mode standard.
onloadwff.js:71  Assertion failed: Input argument is not an HTMLInputElement
getFormProfile @ onloadwff.js:71
setFieldValue @ onloadwff.js:71
formKeydownListener @ onloadwff.js:71
Show 3 more frames
Show less
Chat.svelte:314 submitPrompt 02fc276d-ae2a-45cd-9249-410da7b04522
Chat.svelte:453 modelId gpt-4o
CallOverlay.svelte:518 Received chat start event for message ID ecb09279-d455-49d7-bbe2-bec818f082a6
/openai/chat/completions:1 
        
        
        Failed to load resource: the server responded with a status of 413 ()
Chat.svelte:1099  SyntaxError: Unexpected token '<', "<html>
<h"... is not valid JSON
P @ Chat.svelte:1099
st @ Chat.svelte:1051
CallOverlay.svelte:506 Audio monitoring and playing stopped for message ID da1475a2-f57f-48bf-8ef7-86bfa1941bc7
Audio.svelte:85 Object
```

**Comment by FelixHuoEZ** [comment]
same issue +1

**Comment by krinlove99** [comment]
same question ，Included in long conversations😂

**Comment by wxjttxs** [comment]
same question when upload a image using my own model

**Comment by jonathangoorin** [comment]
same issue, tried to change the context length to no use.
It happens every time, when there is very long messages.

**Comment by tjbck** [comment]
Unfortunately, nginx misconfiguration issue is out of the scope for this project, help from the community wanted here!

**Comment by EnergeticMrMask** [comment]
I have encountered the same issue while using `Nginx` as a reverse proxy for `OpenWebUI`. As @tjbck mentioned, this is a problem caused by `Nginx`. When using `Nginx` to reverse proxy `OpenWebUI`, if the content you input or the generated model content is too large, this issue may occur. ~The simplest solution is to avoid using Nginx as a reverse proxy~(just kidding). You can modify the `client_max_body_size` in `Nginx`'s configuration, increasing it to a size that won't cause issues or setting it to 0, depending on your needs. After I making this adjustment, the problem no longer occurred.

However, I believe that `Nginx` isn't entirely to blame for this issue. When I used other tools, such as `Lobechat`, this problem did not arise. Based on my understanding, it seems that other tools inherently support context compression, meaning that the content generated during each conversation is manageable. `OpenWebUI`, on the other hand, lacks this feature, which leads to an increase in the amount of generated content as the conversation continues, ultimately triggering Nginx's 413 error code.

I believe that supporting this feature is important for both resolving this issue and conserving computational resources (When I switched from `Lobechat` to `OpenWebUI`, I found that my `OpenAI API` consumption increased by 8 times under the same usage intensity). Although it can currently be resolved using functions, using functions is not convenient. This functionality has been mentioned in other issues, such as #6741,#7363. I hope that `OpenWebUI` can support this feature in the future.

**Comment by rgaricano** [comment]
It also happens with images, as discussed at https://discord.com/channels/1170866489302188073/1325496515304755252

**Comment by cainkrix** [comment]
Not just nginx, I changed from Nginx to Traefik and the same problem still occurs, /api/v1/files/ 413 (Request Entity Too Large), with also errors below,
window.fetch @ fetcher.js:76
c @ index.ts:8
m @ KnowledgeBase.svelte:151
pe @ KnowledgeBase.svelte:600Understand this errorAI
index.ts:22 SyntaxError: Unexpected token '<', "<html>
<h"... is not valid JSON

Anyone knows how to fix the issue?

**Comment by 09Catho** [comment]
I have fixed it, so it's actually nginx config fault by default it supports 1MB so when more than that happens which is often in case of blobs so this happens, its pretty quick fix, go to the nginx config there in server{...} block add  this code `client_max_body_size 50M` you could replace 50M by your preferred size but for me it's a sweet spot, then just `sudo nginx -t sudo systemctl reload nginx
` and hurray it will be fixed now, hope you found it useful.
